src/audio/host.c

```diff
@@ -158,7 +158,8 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 		hd->report_pos += bytes;
 
 		/* send IPC message to driver if needed */
-		if (hd->report_pos >= dev->params.host_period_bytes) {
+		if (hd->report_pos >= dev->params.host_period_bytes &&
+		    dev->params.host_period_bytes != 0) {
```

Comment:
Let's switch the condition to make it more optimized:
```
if (dev->params.host_period_bytes &&
    hd->report_pos >= dev->params.host_period_bytes)
```

Reply:
thanks, I will refine it